Week1/homework/world-queries.txt

```diff
@@ -37,4 +37,181 @@ SELECT city.Name FROM city JOIN country ON city.CountryCode = country.Code WHERE
+37 rows in set (0.00 sec)
+
+4. What's the top 10 countries by SurfaceArea ?
+mysql> SELECT Name FROM country ORDER BY SurfaceArea DESC LIMIT 10;
++--------------------+
+| Name               |
++--------------------+
+| Russian Federation |
+| Antarctica         |
+| Canada             |
+| China              |
+| United States      |
+| Brazil             |
+| Australia          |
+| India              |
+| Argentina          |
+| Kazakstan          |
++--------------------+
+10 rows in set (0.00 sec)
+
+5. What's the top 10 most populated cities?
+mysql> SELECT Name FROM city ORDER BY Population DESC LIMIT 10;
++-------------------+
+| Name              |
++-------------------+
+| Mumbai (Bombay)   |
+| Seoul             |
+| São Paulo         |
+| Shanghai          |
+| Jakarta           |
+| Karachi           |
+| Istanbul          |
+| Ciudad de México  |
+| Moscow            |
+| New York          |
++-------------------+
+10 rows in set (0.00 sec)
+
+6. Select the top 3 countries by population that their names start with the letter ‘P’
+mysql> SELECT Name FROM country WHERE Name LIKE 'P%' ORDER BY Population DESC LIMIT 3;
++-------------+
+| Name        |
++-------------+
+| Pakistan    |
+| Philippines |
+| Poland      |
++-------------+
+3 rows in set (0.00 sec)
+
+7. What is the population in Asia?
+mysql> SELECT SUM(Population) FROM country WHERE Continent = 'Asia';
++-----------------+
+| SUM(Population) |
++-----------------+
+|      3705025700 |
++-----------------+
+1 row in set (0.00 sec)
+
+8. Which languages are spoken in the Region of ‘South America’
+mysql> SELECT Language FROM countrylanguage JOIN country WHERE (countrylanguage.CountryCode = country.Code AND Region = 'South America');
++------------------+
+| Language         |
++------------------+
+| Indian Languages |
+| Italian          |
+| Spanish          |
+| Aimará           |
+| Guaraní          |
+| Ketšua           |
+| Spanish          |
+| German           |
+| Indian Languages |
+| Italian          |
+| Japanese         |
+| Portuguese       |
+| Aimará           |
+| Araucan          |
+| Rapa nui         |
+| Spanish          |
+| Arawakan         |
+| Caribbean        |
+| Chibcha          |
+| Creole English   |
+| Spanish          |
+| Ketšua           |
+| Spanish          |
+| English          |
+| Creole French    |
+| Indian Languages |
+| Arawakan         |
+| Caribbean        |
+| Creole English   |
+| Aimará           |
+| Ketšua           |
+| Spanish          |
+| German           |
+| Guaraní          |
+| Portuguese       |
+| Spanish          |
+| Hindi            |
+| Sranantonga      |
+| Spanish          |
+| Goajiro          |
+| Spanish          |
+| Warrau           |
++------------------+
+42 rows in set (0.00 sec)
+
+9. What are the languages spoken on all cities named ‘Barcelona’
+mysql> SELECT Language FROM countrylanguage JOIN (city) WHERE (countrylanguage.CountryCode = 'ESP' AND city.Name = 'Barcelona');
```

Comment:
While technically correct, you used some information that wasn't given in the question - you provided 'ESP' as the country code, but the question didn't give you that information. Give this one another try - remember, you'll need to involve all three tables.

Reply:
I'm trying to figure out why we need three tables! it might take a bit of time so please bear with me but I've found another way where I don't use the CountryCode explicitly but still only 2 tables.
I'll push the changes again :)